modules/table/src/main/java/org/apache/ignite/internal/table/distributed/replicator/PartitionReplicaListener.java

```diff
@@ -116,9 +139,81 @@ public CompletableFuture<Object> invoke(ReplicaRequest request) {
             return processMultiEntryAction((ReadWriteMultiRowReplicaRequest) request);
         } else if (request instanceof ReadWriteSwapRowReplicaRequest) {
             return processTwoEntriesAction((ReadWriteSwapRowReplicaRequest) request);
+        } else if (request instanceof ReadWriteScanRetrieveBatchReplicaRequest) {
+            return processScanRetrieveBatchAction((ReadWriteScanRetrieveBatchReplicaRequest) request);
+        } else if (request instanceof ReadWriteScanCloseReplicaRequest) {
+            return processScanCloseAction((ReadWriteScanCloseReplicaRequest) request);
         } else {
-            throw  new UnsupportedReplicaRequestException(request.getClass());
+            throw new UnsupportedReplicaRequestException(request.getClass());
+        }
+    }
+
+    /**
+     * Close all cursors connected with a transaction.
+     *
+     * @param txId Transaction id.
+     * @throws Exception When an issue happens on cursor closing.
+     */
+    public void closeAllTransactionCursors(UUID txId) throws Exception {
+        var lowCursorId = new IgniteUuid(txId, Long.MIN_VALUE);
+        var upperCursorId = new IgniteUuid(txId, Long.MAX_VALUE);
+
+        Map<IgniteUuid, Cursor<BinaryRow>> txCursors = cursors.subMap(lowCursorId, true, upperCursorId, true);
+
+        for (Cursor<BinaryRow> cursor : txCursors.values()) {
+            cursor.close();
```

Comment:
Please also add exception handling here in order continue closing rest cursors event if some of them are failed.